pkg/knowledge_base2/properties/map_property.go

```diff
@@ -112,6 +112,16 @@ func (m *MapProperty) Parse(value any, ctx knowledgebase.DynamicContext, data kn
 					continue
 				}
 				result[key] = val
+			} else {
+				val, err := prop.GetDefaultValue(ctx, data)
```

Comment:
i guess it depends if we think that our templates are setup in a manner to support that

in the template below, you wouldnt be able to have the default_value on cookies, it would have to live on the ForwardedValues, otherwise you may never evaluate Cookies
```
      ForwardedValues:
        type: map
        properties:
          QueryString:
            type: bool
            default_value: true
          Cookies:
            type: map
            properties:
              Forward:
                type: string
                default_value: none
```

so the template would have to look like.
```
      ForwardedValues:
        type: map
        default_value:
          Cookies: none
        properties:
          QueryString:
            type: bool
            default_value: true
          Cookies:
            type: map
            properties:
              Forward:
                type: string
                default_value: none
```

Theres the potential for this to cause issues if people use the append command instead of the set command because then they may append certain values that are incompatible (which validation would catch) but there would be no possible way for them to ever initialize ForwardedValues with QueryString= false if that conflicts with Cookies=none.

doesnt our property deps finder run the TransformPropertyValue for the default Value? In which case it should pick up on nested dependencies if theyre templated